mlflow/models/signature.py

```diff
@@ -247,6 +254,36 @@ def _infer_signature_from_type_hints(func, input_arg_index, input_example=None):
     return ModelSignature(inputs=input_schema, outputs=output_schema)
 
 
+def _infer_signature_from_input_example(input_example, wrapped_model):
+    """
+    Infer the signature from an example input and a PyFunc wrapped model. Catches all exceptions.
+
+    :param input_example: An instance representing a typical input to the model.
+    :param wrapped_model: A PyFunc wrapped model which has a `predict` method.
+    :return: A `ModelSignature` object containing the inferred schema of both the model's inputs
+        based on the `input_example` and the model's outputs based on the prediction from the
+        `wrapped_model`.
+    """
+    try:
+        input_schema = _infer_schema(input_example)
+        # Copy the input example so that it is not mutated by predict()
+        prediction = wrapped_model.predict(deepcopy(input_example))
+        # For column-based inputs, 1D numpy arrays likely signify row-based predictions. Thus, we
+        # convert them to a Pandas series for inferring as a single ColSpec Schema.
+        if (
+            not input_schema.is_tensor_spec()
+            and isinstance(prediction, np.ndarray)
+            and prediction.ndim == 1
+        ):
+            prediction = pd.Series(prediction)
```

Comment:
Does `pd.Series` preserve the data type?

Reply:
Yes, since we're casting from numpy array.